crates/nu-command/src/filesystem/open.rs

```diff
@@ -146,13 +146,10 @@ impl Command for Open {
                         }
                     };
 
-                    let content_type = if raw {
-                        path.extension()
-                            .map(|ext| ext.to_string_lossy().to_string())
-                            .and_then(|ref s| detect_content_type(s))
-                    } else {
-                        None
-                    };
+                    let content_type = path
```

Comment:
Makes sense for the files where we don't have a `from ...` converter, but it should probably be reset to `None` if we can succesfully convert it to structured data. Otherwise it treats the internal strings as if they were the particular file format, right?

Reply:
What happens is it passes the decision to `detect_content_type()` which uses `mime_guess::from_ext(extension)` which returns an Option<String>. I don't think having it this way causes any issues because it'll just set the metadata content to some or none. even if I do `ls | to csv | save ls.csv` with the display_output hook, i can still do `open ls.csv` without any issues. the only difference i see so far is that `open ls.csv | metadata` says `content_type: text/csv`.

Reply:
Only the `to ...` commands set the content_type metadata, as the meaning of the content type only makes sense for the string/binary (stream) encoded according to that specific format. 

After `from` the content type doesn't make sense anymore as it is in the format agnostic internal `Value` representation. If `open` acts like `--raw` the data is still formatted with its content type, so it should be set in the `PipelineMetadata`. If it is in `Value` representation I think it shouldn't.

Reply:
The point is to allow you to `open file.json` and have it highlighted by the display_output hook or `open file.nu` and have it highlighted. If you don't set the content-type on `open` then that won't happen.

Reply:
If you `open file.json` there is no text that should be highlighted as json or sent like that to `http post` anymore but a parsed nushell structure. Different behavior than `open --raw file.json` or `open script.nu`.

Reply:
I don't agree with you, but I'll change the --raw part back because you tend to be right about most things. :-)